test/api/stripe_test.go

```diff
@@ -147,6 +151,134 @@ func testStripeCheckoutPOST(t *testing.T, at *test.AccountsTester) {
 	}
 }
 
+// testStripeCheckoutIDGET ensures that we can get the info for a checkout
+// session and act on it, i.e. promote the user, if needed.
+func testStripeCheckoutIDGET(t *testing.T, at *test.AccountsTester) {
+	name := test.DBNameForTest(t.Name())
+	// Create a test user.
+	r, _, err := at.UserPOST(name+"@siasky.net", name+"pass")
+	if err != nil {
+		t.Fatal(err)
+	}
+	c := test.ExtractCookie(r)
+	at.SetCookie(c)
+	defer func(c *http.Cookie) {
+		at.SetCookie(c)
+		_, _ = at.UserDELETE()
+	}(c)
+	u, _, err := at.UserGET()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Fixture values.
+	sessionIDWithSub5 := "cs_test_a1fQmmAWGp1woxtWil1Xvx1wtv04fXErpaB7d5avGKvxoZiM86tJeATPZ3"
+	sessionIDWithSub20 := "cs_test_a1fQmmAWGp1woxtWil1Xvx1wtv04fXErpaB7d5avGKvxoZiM86tJeATPZ4"
+	sessionIDWithoutSub := "cs_test_a1fQmmAWGp1woxtWil1Xvx1wtv04fXErpaB7d5avGKvxoZiM86tJeATPZ5"
+	sessionIDInactiveSub := "cs_test_a1fQmmAWGp1woxtWil1Xvx1wtv04fXErpaB7d5avGKvxoZiM86tJeATPZ6"
+	sessionIDPricelessSub := "cs_test_a1fQmmAWGp1woxtWil1Xvx1wtv04fXErpaB7d5avGKvxoZiM86tJeATPZ7"
+	priceID5 := "price_1IReXpIzjULiPWN66PvsxHL4"
+	priceID20 := "price_1IReY5IzjULiPWN6AxPytHEG"
+	stripeID := "cus_M0WOqhLQj6siQL"
+
+	// Set the user's Stripe ID to the one from the fixture.
+	u.StripeID = stripeID
+	// Make sure the StripeID is also updated in the server DB. We can't run a
+	// simple at.DB.UserSave() because the tester and the server might be
+	// running off different databases.
+	// See https://linear.app/skynetlabs/issue/SKY-1239/accounts-tester-parallel-testers
+	_, _, err = at.UserPUT("", "", stripeID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer gock.Off()
+	// We need to enable networking in order to allow the Tester to call our
+	// own API.
+	gock.EnableNetworking()
+	// Set up a response that will upgrade the user to tier 20.
+	gock.New("https://api.stripe.com").
```

Comment:
could be a helper called `mockSessionResponse` 

```go
// mock tier 20 success
res := strings.NewReader(fixtures.StripeCheckoutSessionWithSubTier20)
mockSessionResponse(sessionIDWithSub20, http.StatusOK, res)

// mock tier 20 failure
mockSessionResponse(sessionIDWithSub20, http.StatusOK, nil)
```

Reply:
I usually also move this to right before I'm doing the assertions instead of a big block in the beginning of the test:

```go
// mock tier 20 success
res := strings.NewReader(fixtures.StripeCheckoutSessionWithSubTier20)
mockSessionResponse(sessionIDWithSub20, http.StatusOK, res)

// write tier 20 success assertions here
...

// mock tier 20 failure

// write tier 30 failure assertions here
```